activejob/lib/active_job/arguments.rb

```diff
@@ -69,6 +69,8 @@ def serialize_argument(argument)
           result = serialize_hash(argument)
           result[SYMBOL_KEYS_KEY] = symbol_keys
           result
+        when Proc
```

Comment:
What if the argument is lambda which waits for some parameters?

Reply:
good point, everything what I can suggest is:
```
when Proc
  if argument.parameters.count.zero?
    serialize_argument(argument.call)
  else
    raise SerializationError.new("Unsupported argument type: #{argument.class.name}")
  end
```

may be you have another suggestion